src/main/java/com/ase/restservice/controller/FinanceController.java

```diff
@@ -0,0 +1,79 @@
+package com.ase.restservice.controller;
+
+import com.ase.restservice.exception.ResourceNotFoundException;
+import com.ase.restservice.model.Stock;
+import com.ase.restservice.service.FinanceService;
+import com.ase.restservice.service.StockService;
+import io.swagger.v3.oas.annotations.Operation;
+import java.io.IOException;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Controller for /finance endpoints.
+ */
+@RestController
+public class FinanceController {
+    @Autowired
+    private FinanceService financeService;
+
+    @Autowired
+    private StockService stockService;
+
+    /**
+     * Retrieve a stock's real-time price via Yahoo! Finance API,
+     *
+     * @param stockId StockID
+     * @return Real-time stock price via Yahoo! Finance API
+     * @throws ResourceNotFoundException if stock ID is invalid
+     */
+    @Operation(summary = "Get price of stock given stockId from Yahoo! Finance API")
+    @GetMapping("/finance/{stockId}/price")
+    public Float getApiStockPrice(@PathVariable(value = "stockId") final String stockId)
+            throws ResourceNotFoundException, IOException {
+        return financeService.getStockPrice(stockId);
+    }
+
+    /**
+     * Update all stock prices in database with real-time price.
+     *
+     * @return List of stocks and their updated stock prices in the database
+     * @throws ResourceNotFoundException if stock ID stored in the database is invalid
+     * @throws IOException if connection error to Yahoo! Finance API
+     */
+    @Operation(summary = "Update all stock prices in database with real-time price")
+    @GetMapping("/finance/updateAllStockPrices")
+    public List<Stock> updateAllStockPrices()
```

Comment:
Should we put the logic for updateAllStockPrices into a function in the service? I think this would be more in line with how we have set up other controllers.

Reply:
Ah, yes. I see what you mean. 

The initial idea was to have it as an endpoint so we have a trigger (@boraelci). Now that I manually tested it against the database and see that it works (still warrants unit testing though), I can hide this as service logic (and I think we should)...

But we should have a more detailed discussion about how/when/if we want this logic to be triggered (or do we want to `Updating each stock price as needed (i.e., just before a client or computation requires some information about a stock like a buy or sell)` using [updateStockPrices()](https://github.com/wu-rymd/kaiserschmarrn/pull/36/files#diff-1afc24279e5ff8edecf9b03cc01d867e4afe4c7ee804110f5bde5b1cfb8b9eceR73-R78)?)

Reply:
With that said, @jlamborn324 please advise on next step, thanks!

Reply:
@wu-rymd Sounds good.  As for when to trigger this logic, I think that triggering the updateStockPrices() as needed would be a nice addition, but requires changes to many different parts of the application. I think we could get away with not implementing it if we don't have the time, and we can just discuss in our demo that we update all stock prices on a schedule (once a day or something like that)

Reply:
@jlamborn324 One option was the update all the stock prices on a schedule, and the other was to update one stock price at a time right before it's needed :) But I will close out this PR and merge! Thanks! :D